### ISSUE TITLE
net-train graph issue

### ISSUE THREAD

**Issue by boldjoel** [bertinetto/siamese-fc#15]
Hey, thanks a lot for your work. And I follow your instruction and try to produce my own model for tracking. However, the train graph looks really weird since the errmax is almost 0 from the start of training 
[net-train.pdf](https://github.com/bertinetto/siamese-fc/files/812299/net-train.pdf). I train this model with around 3500 frames from video. Just wonder if you have any clue for the training.

**Comment by bertinetto** [comment]
Hi,
You shouldn't worry about `errmax` when you are not training with negative pairs (which is the default).

The problem here is that it the network it is overfitting really badly, as you can see from the increasing validation error on the loss.

We observed that a bit of overfitting doesn't necessarily means worse performance in tracking. I think this is because the problem the network is trying to minimize (matching a pair of images) is not strictly the same of tracking, where ideally we would like to minimize the overall IOU.
However, the case you presented is overfitting really badly, you should take care of that. I feel the culprit is the small amount of training frames you are using.

Hope this helps.
